Linux/audit.sh

```diff
@@ -79,10 +79,39 @@ harden() {
 }
 
 
+backup_config_dirs() {
+        # Takes 1 argument - Array of strings of paths that need to be backed up.
+
+        # Get paths that exist
+        local -n dir_arr=$1
+        dir_str=""
+
+        for path in ${dir_arr[@]};
+        do
+                [[ -d $path ]] && dir_str="$dir_str $path"
+                [[ -f $path ]] && dir_str="$dir_str $path"
+        done
+
+        [ -z "$dir_str" ] && echo "Could not backup files. No valid paths in array." && return
+
+        # Make backup in /Backups directory in Home
+        mkdir -p ~/Backups
+        version="$(find ~/Backups -name "$HOSTNAME-config-dir-backup-[0-9]*\.tgz" | wc -l)"
+        tar -czf ~/Backups/$HOSTNAME-config-dir-backup-$version.tgz $dir_str
+}
+
+
+
 #below should both be false
 ShouldUpdate=false
 ShouldInstall=false
 
+# To see if this is the first time running the script.
+# Useful for backing up config directories.
+[[ ! -e ./auditlog.log ]] && touch auditlog.log && echo 0 > auditlog.log
+timesRun=$(echo $(head -n 1 "./audit.log") + 1 | bc -l)
```

Comment:
This might be able to roll into the backup version counter, maybe just check to see if there is a config already backed up and then only run the function if there is no backup tar unless it's specified to backup with a command or something.